spec/services/google/parse_service_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Google::ParseService, type: :service do
+  describe '#call' do
+    context 'when parsing a page having 3 top ads and 1 bottom ad' do
+      it 'counts 3 top ads', vcr: 'google/programming_courses_3_top_1_bottom' do
+        html = Google::SearchService.search!('programming courses')
+        top_ads = described_class.new(html).call.search_entries.count { |e| e.kind == :ads && e.position == :top }
+        expect(top_ads).to eq(3)
+      end
+
+      it 'counts 1 bottom ads', vcr: 'google/programming_courses_3_top_1_bottom' do
+        html = Google::SearchService.search!('programming courses')
+        bottom_ads = described_class.new(html).call.search_entries.count { |e| e.kind == :ads && e.position == :bottom }
+        expect(bottom_ads).to eq(1)
+      end
+
+      it 'counts 4 ads in total', vcr: 'google/programming_courses_3_top_1_bottom' do
+        html = Google::SearchService.search!('programming courses')
+        bottom_ads = described_class.new(html).call.search_entries.count { |e| e.kind == :ads }
+        expect(bottom_ads).to eq(4)
+      end
+    end
+
+    context 'when parsing a page having no ads' do
```

Comment:
An improved description would be:

```suggestion
    context 'a page WITHOUT any ads' do
```

Reply:
Fixed in https://github.com/Goose97/google-scraper-ruby/pull/20/commits/9988cc8d24c0bff170a67f8cc38e7b31a6adf90d